pkg/cloud/services/ec2/instances.go

```diff
@@ -183,7 +183,13 @@ func (s *Service) CreateInstance(scope *scope.MachineScope, userData []byte) (*i
 
 		return nil, awserrors.NewFailedDependency("failed to run controlplane, APIServer ELB not available")
 	}
-	if !scope.UserDataIsUncompressed() {
+
+	_, userDataFormat, err := scope.GetRawBootstrapDataWithFormat()
```

Comment:
UserData is already passed to this method, how about passing the format as an input also?

Reply:
Yeah, make sense I think. Perhaps we should introduce some intermediate type for passing around the user data together with format together, to avoid having too many parameters to the functions. I'm not sure where to put such type though.

Reply:
For now I just pass it as a separate parameter. Please let me know if this is acceptable or should I define a type and if so, where.